arcgis-ios-sdk-samples/Scenes/Choose camera controller/ChooseCameraControllerViewController.swift

```diff
@@ -0,0 +1,137 @@
+    func planeSymbolDidLoad() {
+        if let error = planeSymbol.loadError {
+            presentAlert(error: error)
+        } else {
+            planeSymbol.heading = 45
+            cameraControllersBarButtonItem.isEnabled = true
+        }
+    }
+
+    /// Returns a scene with imagery basemap and elevation data.
+    func makeScene() -> AGSScene {
+        let scene = AGSScene(basemap: .imagery())
+
+        // Add base surface to the scene for elevation data.
+        let surface = AGSSurface()
+        let worldElevationServiceURL = URL(string: "https://elevation3d.arcgis.com/arcgis/rest/services/WorldElevation3D/Terrain3D/ImageServer")!
+        let elevationSource = AGSArcGISTiledElevationSource(url: worldElevationServiceURL)
+        surface.elevationSources.append(elevationSource)
+        scene.baseSurface = surface
+
+        return scene
+    }
+
+    /// Returns a graphics overlay containing the plane graphic.
+    func makeGraphicsOverlay() -> AGSGraphicsOverlay {
+        let graphicsOverlay = AGSGraphicsOverlay()
+        graphicsOverlay.sceneProperties?.surfacePlacement = .absolute
+        graphicsOverlay.graphics.add(planeGraphic)
+        return graphicsOverlay
+    }
+
+    /// Returns a controller that allows a scene view's camera to orbit the Upheaval Dome crater structure.
+    func makeOrbitLocationCameraController() -> AGSOrbitLocationCameraController {
+        let targetLocation = AGSPoint(x: -109.929589, y: 38.437304, z: 1700, spatialReference: .wgs84())
+        let cameraController = AGSOrbitLocationCameraController(targetLocation: targetLocation, distance: 5000)
+        cameraController.cameraPitchOffset = 3
+        cameraController.cameraHeadingOffset = 150
+        return cameraController
+    }
+
+    /// Returns a controller that allows a scene view's camera to orbit the plane.
+    func makeOrbitGeoElementCameraController() -> AGSOrbitGeoElementCameraController {
+        let cameraController = AGSOrbitGeoElementCameraController(targetGeoElement: planeGraphic, distance: 5000)
+        cameraController.cameraPitchOffset = 30
+        cameraController.cameraHeadingOffset = 150
+        return cameraController
+    }
+
+    // MARK: - Navigation
+
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        if segue.identifier == "CameraControllersPopover" {
+            guard let controller = segue.destination as? CameraControllerTableViewController else { return }
+            controller.cameraControllerDelegate = self
+
+            let cameraControllers = [makeOrbitLocationCameraController(), makeOrbitGeoElementCameraController(), AGSGlobeCameraController()]
+            controller.cameraControllers = cameraControllers
+
+            if let currentCameraController = sceneView.cameraController, let selectedCameraController = cameraControllers.first(where: { String(describing: type(of: $0)) == String(describing: type(of: currentCameraController)) }) {
```

Comment:
You could just compare the types directly rather than creating a string from them:
```suggestion
            if let currentCameraController = sceneView.cameraController, let selectedCameraController = cameraControllers.first(where: { type(of: $0) == type(of: currentCameraController) }) {
```